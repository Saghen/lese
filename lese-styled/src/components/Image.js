import styled from "styled-components";
import { propertyGenerator } from "../helpers";

// TODO: Implement width and height
const imageProperties = propertyGenerator([
  ["responsive", () => "max-width: 100%;"],
  ["center", () => "object-position: center;"],
  ["cover", () => "object-fit: cover"]
]);

export default styled.img`
  ${imageProperties}
`;
