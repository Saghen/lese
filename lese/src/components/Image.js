import styled from "@emotion/styled";
import { propertyGenerator } from "../helpers";

// TODO: Implement width and height
const imageProperties = propertyGenerator([
  ["responsive", () => "max-width: 100%;"],
  ["center", () => "object-position: center;"],
  ["cover", () => "object-fit: cover"]
]);

export default styled.img`
  ${imageProperties}
`;
