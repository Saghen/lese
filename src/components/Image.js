import styled from "@emotion/styled";
import { propertyGenerator } from "../helpers";

const imageProperties = propertyGenerator([
  ["responsive", () => "max-width: 100%;"],
  ["center", () => "object-position: center;"],
  ["cover", () => "object-fit: cover"]
]);

export default styled.img`
  ${imageProperties}
`;
