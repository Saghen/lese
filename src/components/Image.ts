import styled from "@emotion/styled";
import { propertyGenerator } from "../helpers";

export interface ImageProps {
  responsive?: boolean;
  center?: boolean;
  cover?: boolean;
}

// TODO: Implement width and height
const imageProperties = propertyGenerator<ImageProps>([
  ["responsive", () => "max-width: 100%;"],
  ["center", () => "object-position: center;"],
  ["cover", () => "object-fit: cover"]
]);

export default styled.img<ImageProps>`
  ${imageProperties}
`;
