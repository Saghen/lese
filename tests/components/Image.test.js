import React from "react";
import renderer from "react-test-renderer";
import Image from "../../src/components/Image";
import { matchers } from "jest-emotion";

expect.extend(matchers);

test("image props", () => {
  const tree = renderer.create(
    <Image responsive center cover src="http://example.com/example.png" />
  );
  expect(tree).toMatchSnapshot();
});
