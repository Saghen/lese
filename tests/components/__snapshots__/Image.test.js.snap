// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image props 1`] = `
.emotion-0 {
  max-width: 100%;
  object-position: center;
  object-fit: cover;
}

<img
  className="emotion-0"
  src="http://example.com/example.png"
/>
`;
